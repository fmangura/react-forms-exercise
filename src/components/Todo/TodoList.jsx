import React, { useState } from 'react';
import { v4 as uuidv4 } from 'uuid';
import NewTodoForm from './newTodoForm';
import Todo from './Todo';

const TodoList = () => {
    const [TodoList, setTodoList] = useState([]);

    const addTodo = (todo) => {
        setTodoList(TodoList => [...TodoList, {id:uuidv4(), todo:todo}]);
    }

    const delTodo = (id) => {
        setTodoList(todo => {
            const copyTodo = [...todo];
            return copyTodo.filter(todo => todo.id !== id)
        })
    }

    return (
        <>  
            <div>
                <NewTodoForm addTodo={addTodo} />
            </div>
            <h1>To Do:</h1>
                {TodoList.map(({id, todo}) => <Todo key={id} text={todo} id={id} delTodo={delTodo}/>)}
            <hr />
        </>
    )
}

export default TodoList;