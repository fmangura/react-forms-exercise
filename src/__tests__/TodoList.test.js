import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import '@testing-library/jest-dom'
import TodoList from '../components/Todo/TodoList';

it('renders properly', function() {
    render(<TodoList />);
})

it('Matches snapshot', function() {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
})

it('Correct info are rendering', function() {
    const { getByText } = render(<TodoList />);
    
    expect(getByText('To Do:', { exact: false })).toBeInTheDocument();
})

it('can get new todo', function() {
    const {getByText, getByLabelText} = render(<TodoList />);

    //Checks a todo does not exist
    expect(document.querySelector('.Todo')).not.toBeInTheDocument();

    const textInput = getByLabelText("What to do?", { exact: false });
    const submit = getByText('Submit!');

    fireEvent.change(textInput, { target: { value: "Test Todo" }});
    fireEvent.click(submit);

    //Checks that a box exists after the form is submitted
    expect(document.querySelector('.Todo')).toBeInTheDocument();

})

it('can delete todo', function() {
    const {getByText, getByLabelText} = render(<TodoList />);

    const textInput = getByLabelText("What to do?", { exact: false });
    const submit = getByText('Submit!');

    fireEvent.change(textInput, { target: { value: "Test Todo" }});
    fireEvent.click(submit);

    //Checks that the todo was made
    expect(document.querySelector('.Todo')).toBeInTheDocument();
    const x = getByText('x');
    fireEvent.click(x);

    expect(document.querySelector('.Todo')).not.toBeInTheDocument();
    

})