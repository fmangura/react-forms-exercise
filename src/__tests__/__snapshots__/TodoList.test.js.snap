// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <form>
        <label
          for="Todo"
        >
          What to do?
        </label>
        <input
          id="Todo"
          name="Todo"
          style="width: 500px;"
          type="text"
        />
        <br />
        <button>
          Submit!
        </button>
      </form>
    </div>
  </div>
  <h1>
    To Do:
  </h1>
  <hr />
</DocumentFragment>
`;
